experimental/listener_example_test.go

```diff
@@ -0,0 +1,92 @@
+package experimental_test
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"log"
+	"sort"
+
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/api"
+	. "github.com/tetratelabs/wazero/experimental"
+	"github.com/tetratelabs/wazero/wasi_snapshot_preview1"
+)
+
+// listenerWasm was generated by the following:
+//
+//	cd testdata; wat2wasm --debug-names listener.wat
+//
+//go:embed logging/testdata/listener.wasm
+var listenerWasm []byte
+
+// uniqGoFuncs implements both FunctionListenerFactory and FunctionListener
+type uniqGoFuncs map[string]struct{}
+
+// callees returns the go functions called.
+func (u uniqGoFuncs) callees() []string {
+	ret := make([]string, 0, len(u))
+	for k := range u {
+		ret = append(ret, k)
+	}
+	// Sort names for consistent iteration
+	sort.Strings(ret)
+	return ret
+}
+
+// NewListener implements FunctionListenerFactory.NewListener
+func (u uniqGoFuncs) NewListener(def api.FunctionDefinition) FunctionListener {
+	if def.GoFunc() == nil {
+		return nil // only track go funcs
+	}
+	return u
```

Comment:
here's the trick. since this listener only needs stack parameters, it can return itself after filtering what to listen to.